### DISCUSSION TITLE
Please move infrequently used settings to engineering mode

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#1911] [Ideas]
**Problem**
When a beginner runs into trouble using xDrip, they can easily change a lot of settings they should not touch.

There are documents that describe the recommended settings.  Yet, everyday, someone posts, in facebook, asking how they can get xDrip going again.
<br/>

**Suggested Solution**
It would be better if only settings were available to the user, out of engineering mode, that would be related to user preferences, not settings that are supposed to be always set or always disabled.
<br/>
<br/>
Is any of the following settings needed to be different than shown for your device or for any reason other than testing?  Can we move them to engineering mode with the shown settings as default? 
<br/>
 
| **Bluetooth Settings** | |
|--------------------------|-|
| Turn Bluetooth on | ☑️  |
| Bluetooth Watchdog | ☑️ |
| Use Background Scans | ☑️  |
| Constantly reset Bluetooth | :white_large_square: |
| Transmitter (PL) support | :white_large_square: |
| RFduino support | :white_large_square: |
| xBridge+ Polling Mode | :white_large_square: |

| **Other misc. options**        |                                 |
|--------------------------------|-----------------------|
| Run Collector in foreground | ☑️ |
| Battery Optimization prompt | :white_large_square: |
| Samsung Workarounds | ☑️ |

| **G5/G6 Debug Settings** | |
|------------------------------|-|
| Use OB1 Collector |  ☑️ |
| Fallback to xDrip | :white_large_square: |
| Allow OB1 initiate bonding | ☑️ |
| Authenticate G5 before each read | ☑️ |
| Unbond G5 before each read | :white_large_square: |  

| **Less common settings** | |
|-----------------------------|-|
| Aggressive service restarts |  ☑️ |

**Comment by tolot27** [comment]
Beside showing extended settings only in engineering mode a lot of settings should not be visible if the data source/sync agent is not active. This will reduce the number of visible settings significantly. I'm unsure whether we should treat this as a separate issue. At least the PRs should address this independently.

**Comment by jamorham** [comment]
If we put too much behind engineering mode then it will end up getting enabled too frequently. Some of the options above yes engineering mode is appropriate as we really don't expect users to need to modify the defaults, but for some others we could have an `advanced mode` and add that as a dependency to the setting and then iterate the settings and automatically hide those when that mode is not enabled.

**Comment by Navid200** [comment]
@jamorham I hope I understand correctly:
I will create a new setting deep under other misc options.  I will call it something like "Advanced" for English.
Then, I will make the settings I have identified here to be accessible only if the new advanced setting is enabled.   If the new advanced setting is disabled, the identified settings automatically go back to their default settings.

**Comment by jamorham** [comment]
I think they will always default to their set defaults unless the user has changed them but I think within `Preferences.java` and if advanced mode is disabled then the settings will not be changeable if they  are set to have that as a dependency. But to hide them I think we want something in `Preferences.java` to iterate them, look for the advanced dependency and auto-hide those options. I believe that will work.

**Comment by Navid200** [comment]
@jamorham When you say "hide them", do you mean to make it impossible to change them, or do you mean literally hide them so that they are not visible?

**Comment by tzachi-dar** [comment]
2 comments:
1) I would not move them, but rather have them being disabled/hidden based on the advanced option.
2) Some of the settings can be disabled, for example Transmitter (PL) support or RFduino support. Many of them are things that users have to change. For example, use scanning. Although I realy don't like the current situation. Currently, there is a huge chance that any user will have to change that. And if he does not have to change it right now, there will be a phone update that will make him use that. So, these settings have to be visible.
We might want to have them in a different color, but currently, people have to change things like: "Authenticate G5 before each read". (You are not an experience user if you are touching this, but rather you are a novice user that things don't work for you).

**Comment by Navid200** [comment]
@tzachi-dar I can see the following as the thinking behind this.  But, please let me know if you disagree.

If there are settings that almost no one ever needs but for testing or for extremely unusual situations, we move them to the engineering mode.  Meaning the user will only be able to change them in engineering mode.  One example is using a Dexcom G6 in non-native mode.  There are cases when someone will need that.  But, it is very rare.  

If there are settings that are only needed to be different than the default for a small percentage of the population under normal circumstances, we cannot move those to the engineering mode because then, too many people will need to have engineering mode active all the time, which would defeat the point of the engineering mode.
For settings that fall in that category, we are going to create this new setting.  It is similar to putting xDrip in engineering mode except you won't need a voice command to enable it.  It is called Advanced.
People who need to change such settings need to activate the Advanced setting.  But, not engineering mode.

If there are settings that 10% or more of the population need to change from default, we just leave those alone.  Such settings can be changed with no need to activate either engineering mode or Advanced mode.

All the settings I have identified above need to be used exactly in default by all Dexcom users.

1- Are any of those settings needed to be different for a Libre-user?

2- If you can think of any Libre-specific settings that fit the described Advanced case, please let me know if you like me to include them in the new Advanced category.
